### PR TITLE
8344077: Remove security manager dependency in java.io

### PR DESCRIPTION
**Pull request by bplb**
Expunge the use of the `SecurityManager`, `AccessController`, and the like from the `java.io` package.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344077](https://bugs.openjdk.org/browse/JDK-8344077): Remove security manager dependency in java.io (**Bug** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) Review applies to [a15574a3](https://git.openjdk.org/jdk/pull/22219/files/a15574a37b485053fc16f996659016b78e4c2c11)
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**) Review applies to [a15574a3](https://git.openjdk.org/jdk/pull/22219/files/a15574a37b485053fc16f996659016b78e4c2c11)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22219/head:pull/22219` \
`$ git checkout pull/22219`

Update a local copy of the PR: \
`$ git checkout pull/22219` \
`$ git pull https://git.openjdk.org/jdk.git pull/22219/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22219`

View PR using the GUI difftool: \
`$ git pr show -t 22219`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22219.diff">https://git.openjdk.org/jdk/pull/22219.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22219#issuecomment-2484071584)
</details>

### CODE REVIEW

**Comment by bplb** [comment]
Preliminary testing: the `jdk_io` tests pass on Linux and Windows.

File: src/java.base/share/classes/java/io/FilePermission.java [hunk 1]
```diff
@@ -181,8 +181,7 @@ public final class FilePermission extends Permission implements Serializable {
     private static final java.nio.file.FileSystem builtInFS =
         DefaultFileSystemProvider.theFileSystem();
 
-    private static final Path here = builtInFS.getPath(
-            GetPropertyAction.privilegedGetProperty("user.dir"));
+    private static final Path here = builtInFS.getPath(System.getProperty("user.dir"));
```
Comment (RogerRiggs) [inline src/java.base/share/classes/java/io/FilePermission.java:184]: Could be updated to use StaticProperty:
```suggestion
    private static final Path here = builtInFS.getPath(jdk.internal.util.StaticProperty.userDir());
```

**Review by RogerRiggs — APPROVED** [review]
Looks good

File: src/java.base/share/classes/java/io/Console.java [hunk 2]
```diff
@@ -673,23 +671,18 @@ private static Console instantiateConsole() {
...
+                    if (jc != null) {
+                        c = new ProxyingConsole(jc);
                     }
-                    return null;
+                    break;
                 }
-            });
+            }
         } catch (ServiceConfigurationError _) {
             c = null;
```
Comment (naotoj) [inline src/java.base/share/classes/java/io/Console.java:687]: This assignment can be removed

**Review by naotoj — COMMENTED** [review]
`Console` changes look good to me, with a minor suggestion

File: src/java.base/share/classes/java/io/Console.java [hunk 3]
```diff
@@ -661,7 +659,7 @@ public Console console() {
 
     @SuppressWarnings("removal")
```
Comment (naotoj) [inline src/java.base/share/classes/java/io/Console.java:660]: This too

File: src/java.base/share/classes/java/io/FilePermission.java (hunk 1 above)
Comment (bplb) [reply src/java.base/share/classes/java/io/FilePermission.java]: So changed in c7ba335.

File: src/java.base/share/classes/java/io/Console.java (hunk 2 above)
Comment (bplb) [reply src/java.base/share/classes/java/io/Console.java]: Removed in c7ba335.

File: src/java.base/share/classes/java/io/Console.java (hunk 3 above)
Comment (bplb) [reply src/java.base/share/classes/java/io/Console.java]: Removed in c7ba335.

**Review by naotoj — APPROVED** [review]

**Comment by bplb** [comment]
The SCTP changes in db92e04 should not be in this PR. A Git error on my part; will remove.

**Review by naotoj — APPROVED** [review]

**Review by LanceAndersen — APPROVED** [review]

File: src/java.base/share/classes/java/io/FileInputStream.java [hunk 4]
```diff
@@ -131,11 +131,6 @@ public FileInputStream(String name) throws FileNotFoundException {
...
     public FileInputStream(File file) throws FileNotFoundException {
         String name = (file != null ? file.getPath() : null);
-        @SuppressWarnings("removal")
-        SecurityManager security = System.getSecurityManager();
-        if (security != null) {
-            security.checkRead(name);
-        }
         if (name == null) {
             throw new NullPointerException();
         }
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/io/FileInputStream.java:136]: I assume (old) L133-139 can be replaced with `String name = file.getPath()`, the explicit check that name is null goes away with the SM check.

File: src/java.base/share/classes/java/io/FileOutputStream.java [hunk 5]
```diff
@@ -200,11 +200,6 @@ public FileOutputStream(File file, boolean append)
...
     {
         String name = (file != null ? file.getPath() : null);
-        @SuppressWarnings("removal")
-        SecurityManager security = System.getSecurityManager();
-        if (security != null) {
-            security.checkWrite(name);
-        }
         if (name == null) {
             throw new NullPointerException();
         }
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/io/FileOutputStream.java:205]: Same comment as the FOS constructor.

**Review by AlanBateman — COMMENTED** [review]
Looks fine, just re-check the FIS/FOS constructors as original check file==null was for the SM usage.

**Review by AlanBateman — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Comment by bplb** [comment]
Thanks, everyone, for the comments and reviews. I will integrate once this has cleared tier 1-3 testing.

**Pull request closed without merging**